src/main/java/jenkins/plugins/slack/webhook/WebhookHandler.java

```diff
@@ -0,0 +1,139 @@
+package jenkins.plugins.slack.webhook;
+
+
+import jenkins.model.Jenkins;
+
+import hudson.model.Build;
+import hudson.model.Result;
+import hudson.model.Project;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+
+import hudson.security.ACL;
+
+import java.io.IOException;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import jenkins.plugins.slack.webhook.model.SlackPostData;
+import jenkins.plugins.slack.webhook.model.SlackTextMessage;
+import jenkins.plugins.slack.webhook.model.SlackWebhookCause;
+
+import org.acegisecurity.context.SecurityContext;
+import org.acegisecurity.context.SecurityContextHolder;
+
+
+
+public class WebhookHandler {
+    private SlackPostData data;
+
+    public WebhookHandler(SlackPostData data) {
+        this.data = data;
+    }
+ 
+    public SlackTextMessage getProjectLog(String projectName,
+        String buildNumber) {
+
+        SecurityContext ctx = ACL.impersonate(ACL.SYSTEM);
+
+        Project project =
+            Jenkins.getInstance().getItemByFullName(projectName, Project.class);
+
+        if (project == null) {
+            SecurityContextHolder.setContext(ctx);
+            return new SlackTextMessage("Could not find project ("+projectName+")\n");
+        }
+
+        AbstractBuild build =
+            project.getBuildByNumber(Integer.parseInt(buildNumber));
+
+        if (build == null) {
+            SecurityContextHolder.setContext(ctx);
+            return new SlackTextMessage("Could not find build #"+buildNumber+" for ("+projectName+")\n");
+        }
+
+        SecurityContextHolder.setContext(ctx);
```

Comment:
Looking better. You could wrap all of this in a try/finally and reset the context in the finally so that it's in 1 place and is less likely to be broken by future changes.